### PR TITLE
libbpf-tools: add CO-RE filelife

### PR DESCRIPTION
**Pull request by ethercflow**
Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### CODE REVIEW

File: libbpf-tools/filelife.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#define __FILELIFE_H
+
+#define DNAME_INLINE_LEN 32
+#define TASK_COMM_LEN    16
+
+struct event {
+	char file[DNAME_INLINE_LEN];
+	char task[TASK_COMM_LEN];
+	__u64 delta_ns;
+	pid_t tgid;
```
Comment (anakryiko) [inline libbpf-tools/filelife.h]: move these two up? char arrays have 1-byte alignment so best put at the end; fields should generally be sorted by alignment size, not field size

File: libbpf-tools/filelife.c [hunk 2]
```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2020 Wenbo Zhang
+//
+// Based on xfsslower(8) from BCC by Brendan Gregg.
```
Comment (anakryiko) [inline libbpf-tools/filelife.c]: copy/paste from xfsslower

File: libbpf-tools/filelife.h (hunk 1 above)
Comment (ethercflow) [reply libbpf-tools/filelife.h]: Oh, got it. I'll fix this. Thank you for point this!

File: libbpf-tools/filelife.c (hunk 2 above)
Comment (ethercflow) [reply libbpf-tools/filelife.c]: Sorry for this mistake.

File: libbpf-tools/filelife.h (hunk 1 above)
Comment (ethercflow) [reply libbpf-tools/filelife.h]: @anakryiko Fixed, I realized the tools I submitted previously had this problem too. I'll fix them in another PR with `get_ktime_ns` to trace_helpers.c then keep converting others another day. Thanks for your continued guidance!

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request merged**